lib/prometheus/client/data_stores/direct_file_store.rb

```diff
@@ -120,16 +120,30 @@ def all_values
                     [k.to_sym, vs.first]
                   end.to_h
 
+                  if @values_aggregation_mode == ALL
+                    #  TODO: Can I do this in a less gross way?
+                    pid = /(\d+)\.bin$/.match(file_path)[1]
```

Comment:
I'd add the label when writing to the file, and then you get it out for free when reading here, no need to do this. 
Basically, add it in `store_key` below

Reply:
Yeah on reflection that seems better. I wasn't sure how I felt about writing out a label that we were injecting, but I can't see any serious downside to it.